testsuite/classlibrary/TestBuffer.sc

```diff
@@ -284,4 +284,46 @@ TestBuffer : UnitTest {
 
 	}
 
+	test_bufferMsgMethodsConvertFloatToInt {
+		// note, no messages will be sent
+		// so 'Server' is irrelevant
+		var buf = Buffer(Server.default, 2048.0, 1.0, 0),
+		copyBuf = Buffer(Server.default, 2048.0, 1.0, 1),
+		msg,
+		failures = Array.new;
+
+		[
+			// [method: [args], indices that should be int]
+			[allocMsg: nil, [2, 3]],
+			[allocReadMsg: ["a", 0.0, 100.0], [3, 4]],
+			[allocReadChannelMsg: ["a", 0.0, 100.0, [0]], [3, 4]],
+			[readMsg: ["a", 0.0, -1.0, 0.0, 0], [3, 4, 5]],
+			[readChannelMsg: ["a", 0.0, -1.0, 0.0, 0, [0]], [3, 4, 5]],
+			[cueSoundFileMsg: ["a", 0.0], [3, 4]],
+			[writeMsg: ["a", "aiff", "int16", -1.0, 0.0, 0], [5, 6]],
+			[setMsg: [0.0, 1], [2]],
+			[setnMsg: [0.0, [1, 2]], [2]],
+			[getMsg: [0.0], [2]],
+			[getnMsg: [0.0, 10.0], [2, 3]],
+			[fillMsg: [0.0, 10.0, 1.0], [2, 3]],
+			[copyMsg: [copyBuf, 0.0, 0.0, 10.0], [3, 5, 6]]
+		].do { |row|
+			try {
+				msg = buf.performList(row[0], row[1]);
+				if(row[2].any { |index| msg[index].isInteger.not }) {
+					failures = failures.add(row[0]);
+					"Buffer '%' expected integers at indices %, produced %\n".postf(
+						row[0], row[2], msg[row[2]].asCompileString
+					);
+				};
+			} { |error|
+				failures = failures.add(row[0]);
+				"Buffer '%' method threw error %\n".postf(
+					row[0], error.errorString
+				);
+			};
+		};
+		this.assert(failures.isEmpty, "Buffer *Msg methods should convert specific arguments to integer");
```

Comment:
this test could be friendlier to read. personally, i always try to make tests as boring-looking as possible -- just setup, call the function, and assert on the results. even if it results in more lines of code, it's your one chance to see what an idiomatic use of the interface looks like, which can also motivate later API revisions.

but if you like doing it this way (which is 100% fine with me), i'd recommend:
- creating a lambda that does the `performList` and checks. that way you have named parameters and are less likely to make programming mistakes indexing the wrong element of `row`
- make a single assert per conversion (i.e., once per result index). then you don't need additional local variables and your assert messages can be as clear as possible. remember that typically, a dev running the tests will see a printout of all the failures at the end of the run, meaning all your `postf`s are going to be lost. we should not use `post` or `postf` _anywhere_ in the test suite.
- remove the try block since it's not relevant to the test

Reply:
Will do on the function and the posts (change to asserts).

If I remove the try, and someone later accidentally breaks something in the Buffer API, would an error in the method being tested crash the entire test suite? I was assuming that crashing the whole test suite would be A Bad Thing.

Reply:
Done.